### PR TITLE
Integrate adaptive pruning manager

### PR DESCRIPTION
**Pull request by cloudnoize**
- Wiring all the replica adaptive pruning components.

- Passing callback for setting primary to replicaimp.

- Passing tho the reconfiguration handlers the adaptive manager object.

- Reorganizing the files.

-  Passing the internal client to the adaptive manager.

### CODE REVIEW

File: kvbc/src/Replica.cpp [hunk 1]
```diff
@@ -474,7 +484,13 @@ Replica::Replica(ICommunication *comm,
...
       secretsManager_{secretsManager},
       blocksIOWorkersPool_((replicaConfig.numWorkerThreadsForBlockIO > 0) ? replicaConfig.numWorkerThreadsForBlockIO
-                                                                          : std::thread::hardware_concurrency()) {
+                                                                          : std::thread::hardware_concurrency()),
+      AdaptivePruningManager_{
+          concord::performance::IntervalMappingResourceManager::createIntervalMappingResourceManager(
+              replicaResources_,
+              std::vector<std::pair<uint64_t, uint64_t>>{
+                  concord::performance::IntervalMappingResourceManager::default_mapping}),
+          20000ms} {
```
Comment (cloudnoize) [inline kvbc/src/Replica.cpp:493]: we'll need to make it configurable

**Review by titusen — APPROVED** [review]
LGTM

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 2]
```diff
@@ -3247,7 +3247,7 @@ void ReplicaImp::onNewView(const std::vector<PrePrepareMsg *> &prePreparesForNew
   metric_current_active_view_.Get().Set(getCurrentView());
   metric_sent_replica_asks_to_leave_view_msg_.Get().Set(0);
 
-  onViewNumCallbacks_.invokeAll(getCurrentView());
+  onViewNumCallbacks_.invokeAll(isCurrentPrimary());
```
Comment (yontyon) [inline bftengine/src/bftengine/ReplicaImp.cpp:3250]: It is correct that currently the other registered method doesn't use the parameter, but I think we will need to either change the callbacks registry to work with boolean or to have a comment here

**Review by yontyon — APPROVED** [review]
LGTM!
Thanks

**Pull request merged**